sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2WriteSupportCheck.scala

```diff
@@ -56,6 +56,11 @@ object V2WriteSupportCheck extends (LogicalPlan => Unit) {
         failAnalysis(s"Delete by condition with subquery is not supported: $condition")
       }
 
+    case UpdateTable(_, _, _, condition) =>
```

Comment:
can we move it to `DataSourceV2Strategy`? This is not a sanity check, this is just current limitation. We may support subquery in the future and it's better to put the current limitation near the implementation.

Let's also move the delete check there too.

Reply:
Done for `UpdateTable`. For `DeleteFromTable`, we have https://github.com/apache/spark/pull/25652, where I plan to make some improvement for DELETE. Will update the delete check there.